### PR TITLE
feat: Implement A2UI streaming parser

### PR DESCRIPTION
**Pull request by nan-yu**
# Description

_Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots._

_List which issues are fixed by this PR. For larger changes, raising an issue first helps reduce redundant work._

## Pre-launch Checklist

- [ ] I signed the [CLA].
- [ ] I read the [Contributors Guide].
- [ ] I read the [Style Guide].
- [ ] I have added updates to the [CHANGELOG].
- [ ] I updated/added relevant documentation.
- [ ] My code changes (if any) have tests.

If you need help, consider asking for advice on the [discussion board].

[CHANGELOG]: ../CHANGELOG.md
[CLA]: https://cla.developers.google.com/
[Contributors Guide]: ../CONTRIBUTING.md
[discussion board]: https://github.com/google/A2UI/discussions
[Style Guide]: ../STYLE_GUIDE.md

### CODE REVIEW

File: agent_sdks/python/src/a2ui/core/parser/streaming.py [hunk 1]
```diff
@@ -0,0 +1,1309 @@
...
+
+      # Consolidated appending logic
+      if messages and messages[-1].a2ui_json is None:
+        messages[-1].a2ui_json = [m]
+      elif messages and isinstance(messages[-1].a2ui_json, list):
+        messages[-1].a2ui_json.append(m)
+      else:
+        messages.append(ResponsePart(a2ui_json=[m]))
+
+  def delete_surface(self, sid: str) -> None:
```
Comment (jacobsimionato) [inline agent_sdks/python/src/a2ui/core/parser/streaming.py]: Should this be a private method etc?

File: agent_sdks/python/tests/core/parser/test_version_handlers.py
Comment (jacobsimionato) [inline agent_sdks/python/tests/core/parser/test_version_handlers.py:1]: It'd be great to have a more comprehensive suit of tests here, e.g. testing parsing of multiple chunks, which are broken up awkwardly with unclosed brackets etc.

Comment (nan-yu) [reply agent_sdks/python/tests/core/parser/test_version_handlers.py:1]: Agreed that we need a comprehensive suite of tests to validate awkward broken chunks. I've added a bunch of tests exactly for those edge cases—like character-by-character streaming and split tags—in `test_streaming_v08.py`. 

The `A2uiVersionHandler` classes here are stateless helpers that supply regex patterns (sniff_metadata) or process fully assembled dictionaries (handle_complete_object).

Let me know if you meant a different suite of tests.

File: agent_sdks/python/src/a2ui/core/parser/streaming.py (hunk 1 above)
Comment (nan-yu) [reply agent_sdks/python/src/a2ui/core/parser/streaming.py]: Thanks. Updated to private.

File: agent_sdks/python/tests/core/parser/test_version_handlers.py
Comment (jacobsimionato) [reply agent_sdks/python/tests/core/parser/test_version_handlers.py:1]: Oh sorry! I just looked at this quickly and totally misunderstood it, because the diff for the file above got collapsed in the UI. Tests look great!

Comment (nan-yu) [reply agent_sdks/python/tests/core/parser/test_version_handlers.py:1]: Thanks!
